include/drivers/image_sensor.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2019 NXP Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_IMG_SENSOR_H_
+#define ZEPHYR_INCLUDE_IMG_SENSOR_H_
+
+#include <zephyr/types.h>
+#include <device.h>
+#include <errno.h>
+#include <drivers/camera_generic.h>
+#include <drivers/display.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define IMAGE_SENSOR0_NAME "zephyr-img-sensor0"
+#define IMAGE_SENSOR1_NAME "zephyr-img-sensor1"
+
+enum img_sensor_effect {
+	IMG_EFFECT_NORMAL,
+	IMG_EFFECT_NEGATIVE,
+};
+
+enum bytes_width {
+	W1B = 1,
+	W2B = 2,
+	W4B = 4
+};
+
+struct img_sensor_reg {
+	u32_t reg;
+	enum bytes_width w_reg;
+	u32_t value;
+	enum bytes_width w_value;
+};
+
+/*Public data placed at the beginning of driver data of device*/
+struct img_sensor_host {
+	/*Camera module associated to*/
+	enum camera_id id;
+	/*I2C device used for configuration*/
+	struct device *i2c;
+	/*GPIO device used for Power control, optional*/
+	struct device *pwr_gpio;
+	/*GPIO pin used for Power control, optional*/
+	u32_t pin;
+	/*GPIO level used for Power control, optional*/
+	int flag;
+};
+
+struct img_sensor_capability {
+	u32_t pixformat_support;
+	u16_t width_max;
+	u16_t height_max;
+};
+
+struct img_sensor_client {
+	u16_t i2c_addr;
+	u32_t sensor_id;
+	enum bytes_width w_sensor_id;
+	u32_t id_reg;
+	enum bytes_width w_id_reg;
+	int contrast_level;
+	int bright_level;
+	float r_gain_db;
+	float g_gain_db;
+	float b_gain_db;
+	u16_t width;
+	u16_t height;
+	enum img_sensor_effect effect;
+	enum display_pixel_format pixformat;
```

Comment:
I'm wondering if it is a good idea to use display API specific enums in image sensor API.
Maybe it would be better to move the pixel formats to a generic header file.

Reply:
I totally agree with you and am planing to create a new header of pixel format, resolution etc for both display and camera.